app/models/piece.rb

```diff
@@ -5,4 +5,27 @@ class Piece < ApplicationRecord
   def icon_path
     "#{color}_#{type}.svg".downcase
   end
+
+  def create_piece
+    @piece = Piece.create(piece_params)
+  end
+
+  def move_to!(x, y)
+    opposing_piece = Piece.find_by(position_x: x, position_y: y)
```

Comment:
Change to `game.pieces.find_by...`

That way, it will only find a piece that belongs to the current game. Otherwise, it could find pieces inside other games.